### DISCUSSION TITLE
Test Explorer runs all test cases instead of selected ones when number of test cases > 2000

### DISCUSSION THREAD

**Discussion by 1892kishan** [nunit/nunit3-vs-adapter#968] [General]
I have a project where when I select more than 2000 test cases to run in Test Explorer of Visual Studio (17.1.1), it runs all test cases instead of selected. If the count is less than 2000 it works fine. Project link https://github.com/1892kishan/Specflowtest 

(To run this project in Visual Studio please download Specflow extension from Extensions > Manage Extensions [How to install](https://docs.specflow.org/projects/getting-started/en/latest/GettingStarted/Step1.html)

Specflow.Test.csproj:
`<PackageReference Include="NUnit" Version="3.13.3" />`
`<PackageReference Include="SpecFlow.Assist.Dynamic" Version="1.4.2" />`
`<PackageReference Include="SpecFlow.NUnit" Version="3.9.58" />`
`<PackageReference Include="SpecFlow.Plus.LivingDocPlugin" Version="3.9.57" />`
`<PackageReference Include="SpecFlow.Tools.MsBuild.Generation" Version="3.9.58" />`
`<PackageReference Include="NUnit.Console" Version="3.15.0" />`
`<PackageReference Include="NUnit3TestAdapter" Version="4.0.0" />`
`<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />`

Note: When I change **NUnit3TestAdapter" Version="4.0.0" to NUnit3TestAdapter" Version="3.17.0"** it works fine and only runs selected test cases.

**Comment by OsirisTerje** [comment]
Yes,  you reach the default limit for a selection.  Selecting test cases causes a very long list of them to be sent to the engine, and that affects performance.  If you **really** need to do that, you can change that limit to any number you like, by setting a runsettings options like described here:  https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#assemblyselectlimit

**Reply by 1892kishan** [reply]
@OsirisTerje  That works. Thank you so much. :)
